lib/eco/treelexer/lexer.py

```diff
@@ -1,3 +1,29 @@
+class RE_OR(object):
+    def __init__(self, lhs, rhs):
+        self.lhs = lhs
+        self.rhs = rhs
+
+    def __repr__(self):
+        return "OR(%s, %s)" % (self.lhs, self.rhs)
+
+    def __eq__(self, other):
+        return type(other) is type(self) and \
```

Comment:
Do you need the `type` checks? At least you could simplify them to `isinstance(other, RE_OR)`?

Reply:
I started to use type instead of isinstance, as it is more precise and possibly faster as it doesn't need to check for superclasses.

Reply:
Good point about performance... But do you even need the `type` checks at all?

Reply:
I think I initally wrote it this way because the `__eq__` method used to be shared among multiple classes before I realised I had to customise the method anyways. So I could just say `type(other) is RE_OR`.

Reply:
Up to you.

Reply:
I do need the type checks as I don't want `RE_OR` and `RE_PLUS` to be equivalent. I don't think it's worth changing it to `type(other) is RE_OR`. Also as it is now it matches the style `__eq__` in `RE_DEFAULT` where the `type(other) is type(self)` makes sense.

Reply:
Fair enough.